samples/CMakeLists.txt

```diff
@@ -21,3 +21,27 @@ endif()
 if (CONFIG_SECURE_BOOT)
   add_child_image(b0 ${CMAKE_CURRENT_LIST_DIR}/bootloader)
 endif ()
+
+# Special handling for the nRF53 when using RPMSG HCI.
+# Automatically include the hci_rpmsg sample as child image, and change
+# the board to be the network core.
+if (CONFIG_BT_RPMSG_NRF53)
+  if ((BOARD STREQUAL nrf5340_dk_nrf5340_cpuappns) OR
+      (BOARD STREQUAL nrf5340_dk_nrf5340_cpuapp))
+
+    set(hci_rpmsg_BOARD
```

Comment:
I'm not sure what you mean. Here we specify the board to be used for a specific child image. The exact board modification (going from **app to **net) is not very general.

Reply:
You're right, but we'll probably need to generalize it to allow users to create their own boards, so doing a regex match .*app -> .*net in a function could be smart to do now, unless I'm missing something (entirely possible).

Reply:
I think this function would have to be rewritten anyway when the next HW that needs this does not comply to the app/net naming convention (maybe that's not even the role of the SoCs).

My thought was that users will have to add these types of snippets. I see that this is an issue down the line (requiring out in tree patches), but this could maybe solved by having a configurable `.cmake` file included in `samples/CMakeLists.txt` which allows downstream users to perform logic like this out of tree?

Reply:
I would think a regex match is simpler, doesn't need to be a function. Anyways, we can fix this later.